compiler/rustc_mir/src/transform/simplify.rs

```diff
@@ -375,15 +376,19 @@ struct UsedLocals {
     increment: bool,
     arg_count: u32,
     use_count: IndexVec<Local, u32>,
+    /// When true var debug info is always preserved. When false the var debug
+    /// info that refers to otherwise unused local variables will be removed.
+    preserve_var_debug_info: bool,
 }
 
 impl UsedLocals {
     /// Determines which locals are used & unused in the given body.
-    fn new(body: &Body<'_>) -> Self {
+    fn new(tcx: TyCtxt<'_>, body: &Body<'_>) -> Self {
         let mut this = Self {
             increment: true,
             arg_count: body.arg_count.try_into().unwrap(),
             use_count: IndexVec::from_elem(0, &body.local_decls),
+            preserve_var_debug_info: tcx.sess.opts.optimize != OptLevel::Aggressive,
```

Comment:
What criteria was used in choosing this? Other compilers generally don't try to preserve any debuginfo after `-O2` which is `OptLevel::Default`. Futhermore, this is a MIR optimisation so it is unclear to me that deriving this from `-O` flag is actually what we want?